### PR TITLE
Display enterprise ID in profile

### LINKED ISSUE #3990: Display enterprise ID in profile
## Description
**- As a:** Super admin (but if too complex all admins)
**- On page:** `/admin/enterprises/ENTERPRISENAME/edit#/primary_details`
**- I want to be able to do:** see the ID of this enterprise

We will be using Zapier a lot, so either we create zaps to have this ID, or we just add it to the enterprise profile.

The ID would only be shown in the PRIMARY DETAIL page, just above the name and would not be editable.

image

## Acceptance Criteria
1. Logged in as an admin I'm able to see the enterprise ID on `admin/enterprises/ENTERPRISENAME/edit#/primary_details`
2. I can't edit this ID.

### PR DESCRIPTION
**Pull request by pacodelaluna**
This ID is used by people to setup Zapier for example. It can also be used to cross-check values.

Closes #3990

We could get this ID from the Web Console or from the API, but we can also just add it here, it costs very less and people can focus on other things.

#### What should we test?
The enterprise profile should be tested to be sure the value is display properly on the Primary Details section.

#### Release notes
- Display enterprise ID on enterprise profile

Changelog Category: Added

### CODE REVIEW

**Comment by pacodelaluna** [comment]
@luisramos0 @mkllnk I have still 2 concerns:
1. Should I add the i18n key in this PR too?
2. Should I restrict this field to Admin only?

And don't hesitate if you have any other remarks.

File: app/views/admin/enterprises/form/_primary_details.html.haml [hunk 1]
```diff
@@ -63,6 +63,13 @@
       %span{ ng: { class: 'availability.toLowerCase()', hide: "checking" } }
         {{ availability }}
         %i{ ng: { class: "{'icon-ok-sign': availability == 'Available', 'icon-remove-sign': availability == 'Unavailable'}" } }
+  .row{ ng: { show: "Enterprise.sells == 'own' || Enterprise.sells == 'any'" } }
```
Comment (luisramos0) [inline app/views/admin/enterprises/form/_primary_details.html.haml]: I'd say if enterprise sells none you also want to have the id anyway.

File: app/views/admin/enterprises/form/_primary_details.html.haml [hunk 2]
```diff
@@ -63,6 +63,13 @@
...
+    .three.columns.alpha
+      = f.label :permalink, t('.ofn_uid')
```
Comment (luisramos0) [inline app/views/admin/enterprises/form/_primary_details.html.haml]: copy paste error "permalink". should be ofn_uid probably

File: app/views/admin/enterprises/form/_primary_details.html.haml [hunk 3]
```diff
@@ -63,6 +63,13 @@
...
+      %div{'ofn-with-tip' => t('.ofn_uid_tip', link: spree.root_url)}
```
Comment (luisramos0) [inline app/views/admin/enterprises/form/_primary_details.html.haml]: copy paste error here probably, url is not needed

File: app/views/admin/enterprises/form/_primary_details.html.haml [hunk 4]
```diff
@@ -63,6 +63,13 @@
...
+        %a= t('admin.whats_this')
+    .six.columns
+      = f.text_field :id, { 'ng-model' => "Enterprise.id", 'ng-model-options' => "{ updateOn: 'default blur', debounce: {'default': 300, 'blur': 0} }", 'ng-disabled' => true }
```
Comment (luisramos0) [inline app/views/admin/enterprises/form/_primary_details.html.haml]: this is wrong, you dont want a text field, you only want to print the id
maybe just
= {{Enterprise.id}}

File: app/views/admin/enterprises/form/_primary_details.html.haml (hunk 2 above)
Comment (luisramos0) [reply app/views/admin/enterprises/form/_primary_details.html.haml]: the t('.ofn_uid') key needs to be added to en.yml

File: app/views/admin/enterprises/form/_primary_details.html.haml (hunk 1 above)
Comment (pacodelaluna) [reply app/views/admin/enterprises/form/_primary_details.html.haml]: Ah yes, I just copied the pattern from `permalink` row. I will remove these options.

File: app/views/admin/enterprises/form/_primary_details.html.haml (hunk 4 above)
Comment (pacodelaluna) [reply app/views/admin/enterprises/form/_primary_details.html.haml]: Right, I will remove all this Angular stack.

**Comment by pacodelaluna** [comment]
@luisramos0 Sorry about the mess, I should be more careful with copy/paste. I fixed all the points you mentioned and add a screenshot below. Tell if any other issues.
Screenshot from 2019-07-03 23-04-09

File: app/views/admin/enterprises/form/_primary_details.html.haml [hunk 5]
```diff
@@ -63,6 +63,13 @@
       %span{ ng: { class: 'availability.toLowerCase()', hide: "checking" } }
         {{ availability }}
         %i{ ng: { class: "{'icon-ok-sign': availability == 'Available', 'icon-remove-sign': availability == 'Unavailable'}" } }
+  .row
+    .three.columns.alpha
+      = f.label :id, t('.ofn_uid')
+      %div{'ofn-with-tip' => t('.ofn_uid_tip')}
+        %a= t('admin.whats_this')
+    .six.columns
+      = f.text_field :id, value: f.object.id, disabled: true
```
Comment (mkllnk) [inline app/views/admin/enterprises/form/_primary_details.html.haml]: I think just displaying the id would be better, like Luis suggested.

File: config/locales/en.yml [hunk 6]
```diff
@@ -705,6 +705,8 @@ en:
           permalink_tip: "This permalink is used to create the url to your shop: %{link}your-shop-name/shop"
           link_to_front: Link to shop front
           link_to_front_tip: A direct link to your shopfront on the Open Food Network.
+          ofn_uid: OFN UID
+          ofn_uid_tip: The uniq id used to identify the enterprise on Open Food Network.
```
Comment (mkllnk) [inline config/locales/en.yml]: In Ruby it's `uniq` but in plain English it's spelled "unique".

**Comment by pacodelaluna** [comment]
Changes applied, tell me if any other issues.

File: config/locales/en.yml (hunk 6 above)
Comment (pacodelaluna) [reply config/locales/en.yml]: I guess I am talking too much to machines ;)

**Comment by RachL** [comment]
Enterprise profile:
image

API result:
image

This is good to go :)
